### ISSUE TITLE
slick-viewport overflow multiple values

### ISSUE THREAD

**Issue by SatanEnglish** [6pac/SlickGrid#307]
@6pac

 With the version release v2.3.23 slick.grid.js i'm having a issue with the overflow.

With the option forceFitColumns = true.
I then reload the data on the grid with more columns the grid becomes larger than the view.

Expected result is that I should have a bottom scroll.
However I get a double overflow value with hidden first.
`<div class="slick-viewport" style="width: 100%; **overflow: hidden auto**; outline: 0px; position: relative; height: 571px;">
`

**Comment by SatanEnglish** [comment]
This might be by design can you let me know thanks

**Comment by 6pac** [comment]
Is it easy to repro based on a repo example? There was one pull I wasn't 100% sure about in this last release.

**Comment by ghiscoding** [comment]
The best would be to test with previous version of the grid which date to July. 
Here's the [commit](https://github.com/6pac/SlickGrid/blob/151c0736dac9d5717073bd51b092e51572d1fe99/slick.grid.js) of the July version. 
I would suggest @SatanEnglish to try with that older version to see if there's any differences first.

**Comment by SatanEnglish** [comment]
Oh god more time I don't have.
I've tried to replicate and found another thing where item should be auto.
I'm going to hope these are related.

IN example-jquery-accordion.html
Add forceFitColumns: true to the options

Then change myGrid01 to width 30%

Result items outside of grid div and no scroll.

In this case we have ON EDGE we have
overflow: auto;
overflow-x: hidden;
overflow-y: auto;

On Chrome we have 
overflow: hidden auto;

**Comment by SatanEnglish** [comment]
@6pac 

Oh I just replicated that on the version from @ghiscoding PR I was working on.

slick.grid.js says  SlickGrid v2.3.

Not sure if that's the same version as the July one.

The one I found this issue on first is release v2.3.23

**Comment by ghiscoding** [comment]
That was an option `alwaysShowVerticalScroll` that I added long time ago which helps the hamburger menu yes. But this option is disabled by default and when that is the case then it uses the settings prior to the new option. So prior to the new option the line was
```js
$viewport.css("overflow-y", options.autoHeight ? "hidden" : "auto");
```
The commit is this [commit](https://github.com/6pac/SlickGrid/commit/e33bfc813bea770dae8c2f205df2f97e2dbbd5ef#diff-7f1ab5db3c0316e19a9ee635a1e2f2d0)

**Comment by SatanEnglish** [comment]
@ghiscoding 
Was asking if the overflow-x was changed during that change.
Guessing the answer is no.

Guess this case may have always been there then.

**Comment by ghiscoding** [comment]
@SatanEnglish you can always go back to the original version of Mleibman to know if it was there from the start. Or the other way to find changes in a file like `slick.grid.js` is to use the `History` button on top.

And to answer your question, the code looks exactly the same from Mleibman original [file](https://github.com/mleibman/SlickGrid/blob/master/slick.grid.js#L261), which is
```js
$viewport.css("overflow-y", options.autoHeight ? "hidden" : "auto");
```
There is no `overflow-y` anywhere in that original file either. 

![history](https://user-images.githubusercontent.com/643976/50037618-4da23280-ffe1-11e8-958d-ee7013a7bfd6.png)

**Comment by SatanEnglish** [comment]
@ghiscoding  its this line i'm interested in the X axis
$viewport.css("overflow-x", options.forceFitColumns ? "hidden" : "auto");

**Comment by SatanEnglish** [comment]
Original has no css line for the "overflow-x"    

```
$viewport = $("<div class='slick-viewport' style='width:100%;overflow:auto;outline:0;position:relative;;'>").appendTo($container);
$viewport.css("overflow-y", options.autoHeight ? "hidden" : "auto");
```
line 260 of before 6pc first commit

**Comment by SatanEnglish** [comment]
I've double checked and if I take out.
$viewport.css("overflow-x", options.forceFitColumns ? "hidden" : "auto");
This fixes the issue. 


hide overflow-x when options.ForceFitColumns = true as per #70
This is what it's added for

**Comment by SatanEnglish** [comment]
@6pac Idea's on this one?
What's the best options for this?
Another option bool?

Thoughts please

**Comment by 6pac** [comment]
I've had a look, and my feeling is that ```overflow-x``` should just be left as 'auto', setting it to 'hidden' is a hack.
It really should be up to the ```forceFitColumns``` option code to set the correct width initially so that there is no need for a scrollbar to appear, even though it's allowed.
Your case is a good example, where the width is set initially, then changes later on. Another case would be where the set column minimum widths make it impossible to fit the columns in the viewport.

That doesn't help though, as I don't have time to comb the code for edge cases with all the browsers. I will try to put together a test page specifically for this so we can all be singing from the same hymn book in future.

For now, I think a new option ```alwaysShowHorizontalScroll``` that overrides the above behaviour would be the best option. This has been committed to the main repo, I'll wait to do a new release, there are a few minor pulls/tweaks to take care of.

**Comment by SatanEnglish** [comment]
@6pac
Thanks i'll have a look at it.
Not sure alwaysShowHorizontalScroll  is the right name for it tho. (Should not ALWAYS show)
allowForceFitHorizontalSroll <- long winded but this cannot get confused that the scroll will show even if the content is not wide enough to need one.

Shorter would be better but hopefully you see my point with the name

**Comment by 6pac** [comment]
Yes, you're right, I'm just very tired! How about ```alwaysAllowHorizontalScroll```

**Comment by SatanEnglish** [comment]
@6pac
Sounds good to me!

**Comment by 6pac** [comment]
Sorry @ghiscoding , going to change the name of the option as above. That'll teach me to shoot from the hip.

**Comment by ghiscoding** [comment]
That's ok, not long to replicate 😉 

I'm on the final round of testing for the frozen grid feature 😀 
I'd be happy to release it before year end

**Comment by SatanEnglish** [comment]
I've got a new version of autocolumnsize plugin coming to.
Seems to be nice now and actually resize columns to the right size.

It's all thanks to this bug that I've had time to make it work nice.
I'll try get some examples and a pull request before xmas ready for the new year.
